rebasehelper/output_tool.py

```diff
@@ -19,216 +19,200 @@
 #
 # Authors: Petr Hracek <phracek@redhat.com>
 #          Tomas Hozza <thozza@redhat.com>
+from __future__ import print_function
 
 import sys
 import os
 import six
-import json
+import pkg_resources
 
-from rebasehelper.exceptions import RebaseHelperError
-from rebasehelper.logger import LoggerHelper, logger, logger_report
+from rebasehelper.logger import logger
 from rebasehelper.results_store import results_store
 
-output_tools = {}
-
-
-def register_output_tool(output_tool):
-    output_tools[output_tool.PRINT] = output_tool
-    return output_tool
+from colors import red, green, yellow
 
 
 class BaseOutputTool(object):
-
     """
-    Class used for testing and other future stuff, ...
-    Each method should overwrite method like run_check
+    Base class for OutputTools.
+    print_cli_summary must be overridden in order to produce different CLI output
     """
 
     DEFAULT = False
+    NAME = 'name'
+    EXTENSION = 'ext'
 
     @classmethod
-    def match(cls, cmd=None, *args, **kwargs):
-        """Checks if tool name matches the desired one."""
-        raise NotImplementedError()
+    def get_report_path(cls, app):
+        return os.path.join(app.results_dir, 'report.' + cls.get_extension())
+
+    @classmethod
+    def get_extension(cls):
+        raise NotImplementedError
+
+    @classmethod
+    def should_use_colors(cls):
+        """Determine whether ansi colors should be used for CLI output"""
+        if os.environ.get('PY_COLORS') == '1':
+            return True
+        if os.environ.get('PY_COLORS') == '0':
+            return False
+        if cls.app.conf.color == 'auto':
+            if (not os.isatty(sys.stdout.fileno()) or
+                    os.environ.get('TERM') == 'dumb' or
+                    sys.platform.startswith('java')):
```

Comment:
This is not supposed to be here. In the original check this is used in conjunction with NT platform, which we don't support anyway.